src/directives/filereader.js

```diff
@@ -0,0 +1,54 @@
+goog.provide('ngeo.filereaderDirective');
+
+goog.require('goog.asserts');
+goog.require('ngeo');
+
+
+/**
+ * This directive is to used on an input file element. When a file is selected
+ * the directive uses the browser `FileReader` API to read the file. The file
+ * content is provided to the directive user through the assignable expression.
+ * Only works for text files (`readAsText` used for reading the file). And does
+ * not work in Internet Explorer 9.
+ *
+ * @example
+ * <input type="file" ngeo-filereader="ctrl.fileContent" />
+ *
+ * @param {angular.$window} $window The Angular $window service.
+ * @return {angular.Directive} Directive Definition Object.
+ * @ngInject
+ */
+ngeo.filereaderDirective = function($window) {
+  goog.asserts.assert('FileReader' in $window);
```

Comment:
I need to change this, as the directive should not throw an exception when the browser doesn't support `FileReader`. Instead it should somehow report to the application code that the browser does not support the `FileReader` API required for the directive to do its job.